scripts/run_e2e_tests.py

```diff
@@ -296,10 +299,10 @@ def run_tests(args):
                     # Although our unit tests always provide unicode strings,
                     # the actual server needs this failsafe since it can output
                     # non-unicode strings.
-                    line = line.decode('utf-8') # pragma: nocover
+                    line = line.encode('utf-8')  # pragma: nocover
```

Comment:
I am a bit confused about how we choose to use decode/encode or vice versa -- this is something that may be worth clarifying in the wiki doc that I mentioned earlier. In particular why are we changing from encode <-> decode and vice versa below -- was it wrong before?

Reply:
Yeah, it was wrong here. I will try to prepare the wiki for this soon.